### PR TITLE
Feat/modal

### PR DESCRIPTION
**Pull request by darintrimillos**
## Description
Add cdr-modal, which is built from generic-modal and @rei/modal. 

- Design hasn't had formal input on the component, therefore there's no design review
- I don't have unit tests for the `handleFocus` method, so the system doesn't like my level of code coverage. 
- Docs are not created as there's a separate docs story in this upcoming sprint.

### Design:
- [ ] Reviewed with designer and meets expectations

### Cross-browser testing:
- [x] Chrome
- [x] Firefox
- [x] Edge
- [x] Safari
- [x] IE11
- [x] iOS
- [x] Android

### Visual regression testing:
- [x] Added/updated backstop tests
- [x] Passes with existing reference images (or failed as expected)

### Unit testing:
- [ ] Sufficient unit test coverage (see unit test best practices for ideas)

### A11y:
- [x] Meets WCAG AA standards

### Documentation:
- [ ] API docs created/updated
- [ ] Examples created/updated

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -133,6 +133,7 @@
     "stylelint-config-recommended-scss": "^3.3.0",
     "stylelint-scss": "^3.6.1",
     "svg-sprite": "^1.5.0",
+    "tabbable": "^4.0.0",
```
Comment (cowills) [inline package.json]: this should be in `dependencies` since it's used directly by a component

File: src/components/modal/styles/CdrModal.scss [hunk 2]
```diff
@@ -0,0 +1,167 @@
+@import '../../../css/settings/index.scss';
+
+$duration: 150ms;
+$smMin: 576px;
+$overlayColor: rgba(0, 0, 0, .4);
+$contentColor: #fff;
+$modalBorderRadius: .5rem;
+
+.modal {
```
Comment (cowills) [inline src/components/modal/styles/CdrModal.scss]: should these be namedspaced with `cdr-modal` like the rest of our components?

File: src/components/modal/styles/CdrModal.scss [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+$dialog-padding-vertical: $cdr-space-one-and-a-half-x; // 2.4rem
+$header-padding-bottom: $cdr-space-one-x; // 1.6rem
+$title-line-height: $cdr-space-two-x; // 3.2rem
+
+// total up internal padding to get max height of text
+$header-height: $header-padding-bottom + $title-line-height;
+$modal-total-padding: #{$dialog-margins-vertical + $header-height + $dialog-padding-vertical * 2};
+$modal-text-max-height: calc(100vh - #{$modal-total-padding});
+
+.pdp-modal {
```
Comment (cowills) [inline src/components/modal/styles/CdrModal.scss]: should remove pdp specific CSS stuff and make it generic

File: src/components/modal/CdrModal.jsx [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+    label: {
+      type: String,
+      required: true,
+    },
+    labelIsTitle: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+    ariaDescribedbBy: {
```
Comment (cowills) [inline src/components/modal/CdrModal.jsx]: is this used anywhere?

File: src/components/modal/CdrModal.jsx [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+    },
+    closeModal: {
+      type: Function,
+      required: true,
+    },
+    label: {
+      type: String,
+      required: true,
+    },
+    labelIsTitle: {
```
Comment (cowills) [inline src/components/modal/CdrModal.jsx]: `showLabel` might make more sense here as a prop name.

File: src/components/modal/CdrModal.jsx [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+    CdrButton,
+    CdrIcon,
+    CdrText,
+  },
+  props: {
+    opened: {
+      type: Boolean,
+      required: true,
+    },
+    closeModal: {
```
Comment (cowills) [inline src/components/modal/CdrModal.jsx]: this prop is really like `onClose` or `@closed` right? Might be better to just emit a `closed` event instead

`<CdrModal :closeModal="someFunction">` vs. `<CdrModal @closed="someFunction"/>`. The latter seems more in line with the rest of our components

File: src/components/modal/CdrModal.jsx [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+      default: null,
+    },
+    overlayClass: String,
+    wrapperClass: String,
+    contentClass: String,
+    animationDuration: {
+      type: Number,
+      default: 300,
+    },
+    width: {
```
Comment (cowills) [inline src/components/modal/CdrModal.jsx]: should this be `size`?

**Comment by lancetharper** [comment]
Are we wanting to alter API? I thought we take contributions as is? 

I personally would rather make updates now, but we should probably discuss this in critique tomorrow.

**Comment by cowills** [comment]
> Are we wanting to alter API? I thought we take contributions as is?
> 
> I personally would rather make updates now, but we should probably discuss this in critique tomorrow.

i am 100% behind altering the API now. Otherwise it might as well just stay in fedpack.

**Comment by darintrimillos** [comment]
> > Are we wanting to alter API? I thought we take contributions as is?
> > I personally would rather make updates now, but we should probably discuss this in critique tomorrow.
> 
> i am 100% behind altering the API now. Otherwise it might as well just stay in fedpack.

I think this is right, but let's discuss further tomorrow.

File: package-lock.json [hunk 8]
```diff
@@ -17116,13 +17121,23 @@
       "dev": true
     },
     "uglify-js": {
+<<<<<<< HEAD
```
Comment (cowills) [inline package-lock.json]: `rm package-lock.json && npm install` should fix this

File: src/components/modal/CdrModal.backstop.js [hunk 9]
```diff
@@ -0,0 +1,4 @@
+module.exports = [{
+  label: 'Modal',
```
Comment (cowills) [inline src/components/modal/CdrModal.backstop.js:2]: add `url: 'http://localhost:3000/#/modal',` here and delete `selectors`

**Comment by cowills** [comment]
@darintrimillos LGTM aside from the package-lock and backstop comments

**Comment by lancetharper** [comment]
I just screen shared this with Fes and he had some concerns about the content scrolling when there isn't content in the header (because it's just a big blank space at the top)

**Comment by darintrimillos** [comment]
> I just screen shared this with Fes and he had some concerns about the content scrolling when there isn't content in the header (because it's just a big blank space at the top)

Met with Fes yesterday and he approved. I think we're good to go, unless there's anything else anyone sees.

**Review by cowills — APPROVED** [review]

**Pull request merged**